### ISSUE TITLE
Wrong version reported

### ISSUE THREAD

**Issue by jgosmann** [grafana/loki#8318]
**Describe the bug**
Since version 2.7.1, the version is incorrectly exported as branch reference and commit hash. Both when invoking `loki-linux-amd64 -version` and when looking at the `loki_build_info` metric.

For version 2.7.2, for example, I see the following:
```
loki, version HEAD-c35554d (branch: HEAD, revision: c35554d09)
  build user:       root@b28453c1a217
  build date:       2023-01-26T14:10:02Z
  go version:       go1.19.5
  platform:         linux/amd64
```

**To Reproduce**
Run `loki-linux-amd64 -version`.

OR: Look at the `loki_build_info` metric.

**Expected behavior**

First line of the `loki-linux-amd64 -version` output should be:

```
loki, version 2.7.2 (branch: HEAD, revision: c35554d09)
```

The `version` label of the `loki_build_info` metric should have the value `2.7.2`.

**Environment:**
 - Infrastructure: virtual server
 - Deployment tool: Ansible downloading the Loki release from assets of the respective GitHub release.

**Screenshots, Promtail config, or terminal output**

```
$ /usr/local/sbin/loki-linux-amd64 -version
loki, version HEAD-c35554d (branch: HEAD, revision: c35554d09)
  build user:       root@b28453c1a217
  build date:       2023-01-26T14:10:02Z
  go version:       go1.19.5
  platform:         linux/amd64
```

**Comment by mytlogos** [comment]
Should already be fixed with #8232 (070e3ed28fea23ae0cf822f44b5a6e81151f7528).
Currently not released (see CHANGELOG.md on master)

**Comment by ashleykleynhans** [comment]
This is fixed in Loki 2.7.3 which was released today.

**Comment by jgosmann** [comment]
Can confirm that it is fixed. :)
